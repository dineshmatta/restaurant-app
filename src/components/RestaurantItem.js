import React, {useState} from 'react';
import { Card, Button } from 'react-bootstrap';
import {RestaurantItemDetail} from './RestaurantItemDetail';


const handleClick = (setModalShow) => {
  setModalShow(true);
}

const handleOnHide = (setModalShow) => {
  setModalShow(false);
}

const RestaurantItem = ({data}) => {

  const [modalShow, setModalShow] = useState(false);

  return (
    <React.Fragment>
      <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src={data.image_url} alt="Restaurant Image" />
        <Card.Body>
          <Card.Title>{data.name}</Card.Title>
          <Card.Text>
            {data.address}
          </Card.Text>
          <Button variant="primary" onClick={handleClick.bind(null, setModalShow)}>More Details</Button>
        </Card.Body>
      </Card>
      <RestaurantItemDetail data={data} show={modalShow} onHide={handleOnHide.bind(null, setModalShow)} />
    </React.Fragment>
  )
}

export { RestaurantItem };