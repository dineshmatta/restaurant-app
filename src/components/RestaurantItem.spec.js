import React from 'react';
import {RestaurantItem} from './RestaurantItem';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const testData = {
      "id": 21307,
      "name": "Scaramouche Restaurant",
      "address": "1 Benvenuto Place",
      "city": "Toronto",
      "state": "ON",
      "area": "Toronto / SW Ontario",
      "postal_code": "M4V 2L1",
      "country": "CA",
      "phone": "4169618011",
      "lat": 43.68207,
      "lng": -79.40041,
      "price": 4,
      "reserve_url": "http://www.opentable.com/single.aspx?rid=21307",
      "mobile_reserve_url": "http://mobile.opentable.com/opentable/?restId=21307",
      "image_url": "https://www.opentable.com/img/restimages/21307.jpg"
    }
  const tree = renderer
    .create(<RestaurantItem data={testData}></RestaurantItem>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});