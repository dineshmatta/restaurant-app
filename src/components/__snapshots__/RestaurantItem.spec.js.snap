// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    alt="Restaurant Image"
    className="card-img-top"
    src="https://www.opentable.com/img/restimages/21307.jpg"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Scaramouche Restaurant
    </div>
    <p
      className="card-text"
    >
      1 Benvenuto Place
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      More Details
    </button>
  </div>
</div>
`;
